targets/wasm_exec.js

```diff
@@ -298,9 +298,22 @@
 
 					// func finalizeRef(v ref)
 					"syscall/js.finalizeRef": (v_ref) => {
-						// Note: TinyGo does not support finalizers so this should never be
-						// called.
-						console.error('syscall/js.finalizeRef not implemented');
+						// Note: TinyGo does not support finalizers so this is only called
+						// for one specific case, by js.go:jsString. and can/might leak memory.
+						const id = mem().getUint32(unboxValue(v_ref), true);
+						// Note that this if is so far seemingly never true. Someone should investigate why.
+						if (this._goRefCounts?.[id] !== undefined) {
```

Comment:
The `unboxValue` returns the string as a byte array. It leads to `id` being `0` in my case.
So I looked around, and the `unboxValue` contained an interesting part of code:
```
const id = v_ref & 0xffffffffn;
return this._values[id];
```

So I tried this:
```suggestion
						// magicValue taken from `unboxValue` function.
						const magicValue = 0xffffffffn;
						const id = v_ref & magicValue;
						// Note that this if is so far seemingly never true. Someone should investigate why.
						if (this._goRefCounts?.[id] !== undefined) {
```
And when I checked `this._ids.size` before and after the `delete`, it really deleted something :D Can it be it? Is it working? Like, I have no idea what I'm doing and what the hex number is for.

Reply:
@prochac I think in this case is is the highest ID allowed.

Also, from a superficial reading of your proposed change, that looks like it could in fact be the correct way to obtain the ID.

Is there anyone who can test this out?

Reply:
@deadprogram I'd be happy to test, but I am a bit lost wich version... The file as seen here https://github.com/tinygo-org/tinygo/blob/99e6e54e5f0/targets/wasm_exec.js works for me without errors, although the console does say `syscall/js.finalizeRef: unknown id 0`, unsure if expected or not...

Reply:
It would be that same file but with this line https://github.com/tinygo-org/tinygo/blob/99e6e54e5f0/targets/wasm_exec.js#L303C7-L303C59

changed to:

```js
						const id = v_ref & 0xffffffffn;
```

as proposed by @prochac

Reply:
@deadprogram seems to work flawlessly, console is silent

Reply:
I added a commit with the @prochac change, but some how it ended up in @ldemailly fork, and also now this PR has been "Checking for ability to merge automatically…" for about 30 minutes. Something went wrong someplace?

I will try to `git rebase` on my local branch to see if I can get this finally merged into `dev`,,,,